src/subpages/list.tsx

```diff
@@ -132,3 +114,34 @@ const Cards: React.FC<CardsProps> = ({
     </div>
   </div>
 );
+
+const BlockList: React.FC<BlockListProps> = ({ blocks, loading, columns }) => {
+  const timeTaken: IColumn = {
+    key: 'columnTimeTaken',
+    name: 'Time Taken',
+    minWidth: 50,
+    maxWidth: 200,
+    ...commonProps,
+    isSortedDescending: true,
+    data: 'string',
+    isPadded: true,
+    onRender: (block: Block, index) => {
+      if (blocks === null) throw Error('blocks is null');
+      if (index === undefined) throw Error('index is null');
+      const beforeBlock = blocks[Math.min(index + 1, blocks.length - 1)];
```

Comment:
```suggestion
     const beforeBlock = blocks.find(({ index }) => index === block.index - 1);
     if (beforeBlock === undefined) return 0;
```

How about this for index accuracy?

Reply:
Both seem to behave in the same way, except for time complexity.  Whereas index operator is random access, `.find()` is O(n) at least.

Reply:
Well, if we can assume `blocks` are always short I believe @x86chi's suggestion would be good.